README.md

```diff
@@ -449,6 +449,68 @@ Note that because this method is not available in the generic `IDataReader` inte
+--------------------------------------------------------------
+
+With version 2.0.18 and later of the .NET connector, you can send
+a [batch of SQL statements](https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/batches-of-sql-statements?view=sql-server-ver15)  (separated by semicolons)
+to be executed in a single request. For example:
+
+```cs
+IDbCommand cmd = conn.CreateCommand();
+                var param = cmd.CreateParameter();
+                param.ParameterName = "MULTI_STATEMENT_COUNT";
+                param.DbType = DbType.Int16;
+                param.Value = 3;
+                cmd.Parameters.Add(param);
+                cmd.CommandText = "select 1; select 2, 3; select 4, 5, 6";
+                IDataReader reader = cmd.ExecuteReader();
+```
+
+To send a batch of statements with the Snowflake .NET connector, you must specify the number of statements in the batch.
+The Snowflake database requires the exact number of statements in order to guard against
+[SQL injection](https://en.wikipedia.org/wiki/SQL_injection>) attacks. 
+
+The next section explains how to specify the number of statements in a batch. 
+
+### Specifying the Number of Statements in a Batch
+
+By default, the Snowflake database expects a .NET connector application to prepare and send a single statement for execution.
+
+You can override this by specifying the number of statements in a batch for a given request or by enabling multiple statements for
+the current session.
+
+To enable multiple statements for the current session or account, alter the session or account, and set the Snowflake [MULTI_STATEMENT_COUNT](https://docs.snowflake.com/en/sql-reference/parameters.html#label-parameters-multi-statement-count) parameter to ``0``, as follows.
+
+```
+alter session set MULTI_STATEMENT_COUNT = 0;
+```
+
+By default, `MULTI_STATEMENT_COUNT` is set to ``1``, which indicates that only one SQL statement can be executed.
+
+### Preparing a Batch of SQL Statements
+
+The .NET connector supports the ability to prepare a batch of SQL statements (e.g. by calling the `SQLPrepare` function). Note
+the following:
+
+- If the statements have
+  [parameters](https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/statement-parameters?view=sql-server-ver15),
+  calling the [SQLNumParams](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlnumparams-function?view=sql-server-ver15)
+  function returns the total number of parameters in all the statements in the batch.
+
+- Column information about the result set (e.g. data returned by `SQLNumResultCols`, `SQLDescribeCol`, `SQLColAttribute` and
+  `SQLColAttributes`) is available when you call `SQLExecute` or `SQLExecDirect`.
+
+  Although some column information is available when you call `SQLPrepare`, the information might not be completely accurate, and
+  subsequent calls to `SQLExecute` or `SQLExecDirect` might provide more accurate information.
```

Comment:
I think the functions like SQLPrepare are specific to ODBC.

I think ODBC and the .NET connector have very different APIs.

I don't think you can just copy and paste the ODBC documentation into the .NET Connector documentation.

I would check with the engineers on this.

Reply:
The engineers approved this content, but I will touch base to verify.

Reply:
As an example, in .NET, it looks like `DbCommand` has a `Prepare` method (not an SQLPrepare function) for preparing SQL statements:

https://learn.microsoft.com/en-us/dotnet/api/system.data.common.dbcommand.prepare?view=net-6.0

It looks like we don't actually implement that method:

https://github.com/snowflakedb/snowflake-connector-net/blame/e9673a50a3c69bb6a2c7807cf426699b253df9df/Snowflake.Data/Client/SnowflakeDbCommand.cs#L232